providers/refresh.md

```diff
@@ -0,0 +1,140 @@
+## Provider Refresh
+
+Of all of the functions that a provider implements, refresh is one of the most critical.  Inventory is the basis for almost every other function that MIQ offers, be it provisioning, metrics, event-condition-action, reporting all are based on inventory collected from providers.
+
+##### EmsRefresh
+
+A refresh is initiated by a requester through the [`EmsRefresh`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/ems_refresh.rb) class by passing a list of targets (what to be refreshed) to the [`.queue_refresh`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/ems_refresh.rb#L32-L47) method.  The core EmsRefresh class is then responsible for splitting targets up by their ExtManagementSystem [here](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/ems_refresh.rb#L37-L44) and initiating the appropriate provider Refresher class with the list of targets [here](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/ems_refresh.rb#L92-L94).
+
+###### What is a full vs a targeted refresh?
+
+Refresh targets define the scope of what should be collected.  Depending on the targets refreshes are classified into two categories:
+1. Full refresh
+2. Targeted refresh
+
+In a full refresh the top-level Manager is the target and everything will be collected.  A targeted refresh by contrast will only collect inventory related to the specific target(s) greatly reducing the amount of work that has to be done.
+
+Think about creating a new VM, you don't need to ask for every VM again if you could simply ask the provider about that one new VM and its disks/nics/etc...
+
+##### Refresher
+
+Once in the [`Refresher#refresh`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L27-L68) method the provider has the opportunity to perform some pre and post-processing on the targets if they wish, but the bulk of the work is done by the [`Refresher#refresh_targets_for_ems`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L75-L106) method.
+
+Once here the refresh of a target consists of three steps:
+1. Collection of inventory from the native provider: [`#collect_inventory_for_targets`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L84)
+2. Parsing of the inventory into the MIQ schema: [`#parse_targeted_inventory`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L92)
+3. Saving inventory to the database: [`#save_inventory`](https://github.com/ManageIQ/manageiq/blob/jansa-1-beta1/app/models/manageiq/providers/base_manager/refresher.rb#L96)
```

Comment:
I think we should use the term "Persisting" the inventory to the database, to align with the terminology the code uses.